### ISSUE TITLE
Add `MEMPOOL` rule

### ISSUE THREAD

**Issue by lehins** [IntersectMBO/cardano-ledger#4622]
This `MEMPOOL` rule will be called by the `LEDGER` rule, regardless of the `isValid` flag.

It will only be called if the `LedgerEnv` will have a new flag `leForMempool :: Bool` set to `True`

This will rule will have a generic predicate failure:
```
newtype MempoolPredFailure era = MempoolPredFailure Text
```
that has a free form description of a failure.

Most of the time this rule will be empty, but because it will only be invoked on transactions that are being placed into the mempool it does not require a hardfork to have its behavior amended at any point in time.

In other words when `LEDGERS` rule will invoke `LEDGER` then `leForMempool` will be set to `False` and the `MEMPOOL` will not be invoked, thus preventing any change in Ledger behavior that could be induced by this new rule.

We will thus be able to use this rule as a sanitizer of undesired transactions with any node release without requiring a hard fork

**Comment by lehins** [comment]
> We will thus be able to use this rule as a sanitizer of undesired transactions with any node release without requiring a hard fork

It was brought to my attention on a couple different occasions that people have a problem with the choice of word "sanitizes"

For every person out there that has no idea how Ledger works I will provide an explanation of the wording:

MEMPOOL rule is designed to "sanitize" potentially **_malicious_** or **_dangerous_** transactions! We have employed the same mechanism before the MEMPOOL rule was even created, so this is not new.

Moreover, every single ledger rule that does transaction validation has the same purpose as the MEMPOOL rule: reject invalid transaction.
The choice of word "sanitizer" was used, because unlike regular rules, MEMPOOL does not prevent  100% of transactions from landing on chain. In other words there is still a chance for some transactions that would have been rejected by MEMPOOL rule to go through and be added to the chain. In that sense "sanitizing" is weaker then "rejection", and that was the only reason for choosing that word.

The work that we do in Ledger and in Core Tech in general is to keep Cardano safe and reliable. So, I encourage people to avoid speculation that there is some sort hidden agenda in Ledger to censor transactions.
